### DISCUSSION TITLE
Custom-sandbox-command is executed in machine-root `/` instead of project/workspace root

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [ocamllabs/vscode-ocaml-platform#1174] [General]
Type: <b>Bug</b>

When creating a new VScode instance rooted at a particular folder, I expect a 'custom sandbox' command of the form `opam exec $prog -- $args` to function.

Unfortunately, it appears that the CWD is the machine root, `/`, instead of the project-root … so `opam` can't find binaries installed in the project-local switch.

Reproduction:
1. Create a local-sandbox in a new folder; install `ocamllsp` via opam
2. Use `code .` from that folder to lauch VScode
3. Attempt a custom-sandbox config like the following: `pwd; opam exec $prog -- $args`

For me, the logs demonstrate the following error:

```
external command (finished) {
  "shell": "pwd; opam exec ocamlc -- --version",
  "result": {
    "exitCode": 0,
    "stdout": "/\n4.13.1\n",
    "stderr": ""
  }
}

external command (finished) {
  "shell": "pwd; opam exec ocamllsp -- --version",
  "result": {
    "exitCode": 127,
    "stdout": "/\n",
    "stderr": "[ERROR] Command not found 'ocamllsp'\n"
  }
}
```

Extension version: 1.10.7
VS Code version: Code 1.71.0 (784b0177c56c607789f9638da7b6bf3230d47a8c, 2022-09-01T07:25:25.516Z)
OS version: Darwin arm64 21.6.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|3, 4, 7|
|Memory (System)|16.00GB (0.30GB free)|
|Process Argv|--crash-reporter-id ddb5da65-4654-412e-9dc3-738998ade011|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by sloane-shark** [comment]
This is also a problem for me, as my project tooling is specific to my project cwd, not installed globally on my system. A potential workaround would be to expose [VS Code variables](https://code.visualstudio.com/docs/editor/variables-reference) in the custom sandbox syntax so that one could do something like `my_build_tool exec --cwd ${workspaceFolder} --command $prog -- $args`

**Comment by jonahbeckford** [comment]
I very much would like to use the VS Code variables support. Specifically the following two:
- `${workspaceFolder}`
- `${pathSeparator}`
